app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting.tsx

```diff
@@ -20,40 +19,63 @@ export const loader: LoaderFunction = async ({ params, request }: DataFunctionAr
 };
 
 export const action: ActionFunction = async ({ request, params }): Promise<Response> => {
-  const { catchCertificate } = params;
-  const _redirect = `/create-catch-certificate/${catchCertificate}/what-are-you-exporting`
   const form = await request.formData();
-  const { _action, ...values } = Object.fromEntries(form);
-
+  const { catchCertificate } = params;
+  const { _action, faoName, ...values } = Object.fromEntries(form);
+  const _redirect = `/create-catch-certificate/${catchCertificate}/what-are-you-exporting`;
+  
   if (_action === 'cancel') {
     return redirect(`${_redirect}#add-products`);
   }
 
-  const resquestBody: any = {
-    addToFavourites: false,
-    btn_submit:"",
-    presentationLabel:"",
+  if (_action === 'addSpecies') {
+    const errors: IError[] = validateValues(['species'], values);
+    if (errors.length > 0) {
+      return apiCallFailed(errors, values);
+    }
+
+    return redirect(`${_redirect}?species=${values.species}#add-state`);
+  }
+
+  if (_action === 'addState') {
+    const errors: IError[] = validateValues(['species', 'state'], values);
+    if (errors.length > 0) {
+      return apiCallFailed(errors, values);
+    }
+
+    return redirect(`${_redirect}?species=${values.species}&state=${values.state}#add-presentation`)
+  }
+
+  if (_action === 'addPresentation') {
+    const errors: IError[] = validateValues(['species', 'state', 'presentation'], values);
+    if (errors.length > 0) {
+      return apiCallFailed(errors, values);
+    }
+
+    return redirect(`${_redirect}?species=${values.species}&state=${values.state}&presentation=${values.presentation}#add-commodity-code`)
+  }
+
+  const requestBody: any = {
+    addToFavourites: false, // TODO - adding product to favourites
+    commodity_code_description: await getCommodityCodeDescription(values.species as string, values.state as string, values.presentation as string, values.commodity_code as string),
     redirect: _redirect,
-    species:"",
-    stateLabel:"",
-    scientificName:"",
-    commodity_code_description:"",
-    ...values
+    speciesCode: values.species,
+    ...values,
+    species: `${faoName} (${values.species})`
   };
   
-  const response = await addSpecies(catchCertificate, resquestBody);
+  const response = await addSpecies(catchCertificate, requestBody);
   const errors: IError[] = response.errors || [];
 
   if (errors.length > 0) {
-    return apiCallFailed(errors);
+    return apiCallFailed(errors, values);
   }
 
-  // go to the next page but using "/" for now
-  return redirect("/");
+  return redirect(`/create-catch-certificate/${catchCertificate}/what-are-you-exporting`);
 }
 
 const AddSpeciesPage = () => {
-  const { errors = {} } = useActionData() || {};
+  const { errors = {}, species: _faoCode} = useActionData() || {};
```

Comment:
I'm not a big fan of "_name". This is usually used for internal use. In reality in our case is because is coming from a serverReponse or form so I suggest to prefix with something better. something like apiFaoCode or similar

Reply:
Understood, now using `apiFaoCode` @daniele-zurico